crypto/bls/signature_batch.go

```diff
@@ -33,12 +45,58 @@ func (s *SignatureBatch) Verify() (bool, error) {
 	return VerifyMultipleSignatures(s.Signatures, s.Messages, s.PublicKeys)
 }
 
+// VerifyVerbosely verifies signatures as a whole at first, if fails, fallback
+// to verify each single signature to identify invalid ones.
+func (s *SignatureBatch) VerifyVerbosely() (bool, error) {
+	valid, err := s.Verify()
+	if err != nil || valid {
+		return valid, err
+	}
+
+	// if signature batch is invalid, we then verify signatures one by one.
+
+	var sb strings.Builder
+
+	for i := 0; i < len(s.Signatures); i++ {
+		sig := s.Signatures[i]
+		msg := s.Messages[i]
+		pubKey := s.PublicKeys[i]
+
+		valid, err := VerifySignature(sig, msg, pubKey)
+		if !valid {
+			var desc string
+			if len(s.Descriptions) != len(s.Signatures) {
```

Comment:
These checks are going to be ugly and a source of bugs, it would be much better to always have the same number of descriptions and signatures in a set.